src/bokeh/embed/bundle.py

```diff
@@ -177,19 +178,21 @@ def bundle_for_objs_and_resources(objs: Sequence[Model | Document] | None,
     else:
         raise ValueError(f"expected Resources or a pair of optional Resources, got {resources!r}")
 
-    from copy import deepcopy
+    all_objs = _all_objs(objs) if objs else []
 
     # XXX: force all components on server and in notebook, because we don't know in advance what will be used
-    use_widgets = _use_widgets(objs) if objs else True
-    use_tables  = _use_tables(objs)  if objs else True
-    use_gl      = _use_gl(objs)      if objs else True
-    use_mathjax = _use_mathjax(objs) if objs else True
+    use_widgets = _use_widgets(all_objs) if all_objs else True
+    use_tables  = _use_tables(all_objs)  if all_objs else True
+    use_gl      = _use_gl(all_objs)      if all_objs else True
+    use_mathjax = _use_mathjax(all_objs) if all_objs else True
```

Comment:
The logic should be as follows:
```py
if objs is not None:
    all_objs = _all_objs(objs)
    use_widgets = _use_widgets(all_objs)
    use_tables  = _use_tables(all_objs)
    use_gl      = _use_gl(all_objs)
    use_mathjax = _use_mathjax(all_objs)
else:
    use_widgets = True
    use_tables  = True
    use_gl      = True
    use_mathjax = True
```

Reply:
Hmmm. Happy to change it, just for me to clarify: Is there really a functional difference, or do you think it's just cleaner that way?

Reply:
(Note that I had to change the test and call to `_bundle_extensions` as well, please check if that makes sense)

Reply:
The difference is in handling `[]` and `None`. `[]` should result in `False`, whereas `None` in `True`. It's a minor if not trivial difference, but still. That's also why I think it's clearer this way.

Reply:
Got it, thanks!